sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveUDFs.scala

```diff
@@ -235,6 +210,39 @@ private[hive] case class HiveGenericUDF(
   }
 }
 
+class HiveGenericUDFEvaluator(
+    funcWrapper: HiveFunctionWrapper, children: Seq[Expression])
+  extends HiveInspectors
+  with Serializable {
+
+  @transient
+  private lazy val function = funcWrapper.createFunction[GenericUDF]()
+
+  @transient
+  private[hive] val getUDFType = function.getClass.getAnnotation(classOf[HiveUDFType])
```

Comment:
This is only called in the expression, we can move it out

Reply:
In this way, we will have to expose `function`, Is that OK?